src/equisolve/numpy/models/linear_model.py

```diff
@@ -207,9 +216,12 @@ def predict(self, X: TensorMap) -> TensorMap:
         if self.coef_ is None:
             raise ValueError("No weights. Call fit method first.")
 
-        return dot(X, self.coef_)
+        self.coef_ = dict_to_tensor_map(self.coef_)
+        y_pred = dot(X, self.coef_)
+        self.coef_ = tensor_map_to_dict(self.coef_)
+        return y_pred
 
-    def score(self, X: TensorMap, y: TensorMap, parameter_key: str) -> List[float]:
+    def score(self, X: TensorMap, y: TensorMap, parameter_key: str) -> List[float]: # COMMENT why does it return list of floats if we just allow one paramater_key?
```

Comment:
Sorry this is wrong. It only retrurns a single float.

```suggestion
    def score(self, X: TensorMap, y: TensorMap, parameter_key: str) -> float:
```